vortex-fastlanes/src/bitpacking/mod.rs

```diff
@@ -105,11 +102,11 @@ impl Array for BitPackedArray {
         }
 
         if start > self.len() {
-            return Err(VortexError::OutOfBounds(start, 0, self.len()));
+            vortex_bail!(start, 0, self.len());
```

Comment:
I like almost everything else here... but this is suuuuper weird? For these more structured errors, can we require explicit error type?

Reply:
e.g. 
```
 vortex_err!(OutOfBounds: start, 0, self.len())
 ```